pkg/resources/secret_with_basic_authentication.go

```diff
@@ -0,0 +1,163 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var secretBasicAuthenticationSchema = func() map[string]*schema.Schema {
+	secretBasicAuthentication := map[string]*schema.Schema{
+		"username": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Specifies the username value to store in the secret when setting the TYPE value to PASSWORD.",
+		},
+		"password": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Specifies the password value to store in the secret when setting the TYPE value to PASSWORD.",
+		},
+	}
+	return helpers.MergeMaps(secretCommonSchema, secretBasicAuthentication)
+}()
+
+func SecretWithBasicAuthentication() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateContextSecretWithBasicAuthentication,
+		ReadContext:   ReadContextSecretWithBasicAuthentication,
+		UpdateContext: UpdateContextSecretWithBasicAuthentication,
+		DeleteContext: DeleteContextSecretWithBasicAuthentication,
+
+		Schema: secretBasicAuthenticationSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Description: "Secret with Basic Authentication where Secrets Type attribute is set to PASSWORD.",
```

Comment:
Add a link to the docs - check other resources. This concerns other secret resources, too.

Reply:
Added for every secret resource